pkg/download/downloader.go

```diff
@@ -92,6 +107,38 @@ func extractZIP(targetDir string, read io.ReaderAt, size int64) error {
 	return nil
 }
 
+func isSymlinkAllowed(targetDir, oldname, newpath string) error {
+	// no symlinks to absolute paths
+	if filepath.IsAbs(oldname) {
+		glog.V(4).Infof("invalid absolute symlink %v -> %v\n", oldname, newpath)
+		return errors.New("invalid symlink referencing an absolute path in tar")
+	}
+
+	oldpath := filepath.Join(filepath.Dir(newpath), filepath.Clean(oldname))
+	fmt.Printf("LINK %s %s\n", oldpath, newpath)
+
+	if _, isNonEscaping := pathutil.IsSubPath(targetDir, oldpath); !isNonEscaping {
+		glog.V(4).Infof("invalid escaping symlink %v -> %v\n", oldname, newpath)
+		return errors.New("invalid symlink referencing a parent path in tar")
+	}
+
+	return nil
+}
+
+// make a symlink from oldname to newname within the given enclosing targetDir
```

Comment:
nit: GoDoc convention requires first word of the comment is the name of the method.